File: rag_ui/frontend/src/types/history.ts
// History endpoint response shapes

export interface HistoryMessage {
  id: string;
  role: 'user' | 'assistant';
  content: string;
  timestamp: string;
}

export interface Session {
  session_id: string;
  title: string;
  created_at: string;
  updated_at: string;
  message_count: number;
}

export interface SessionsResponse {
  sessions: Session[];
}

export interface SessionDetail {
  session_id: string;
  title: string;
  messages: HistoryMessage[];
}

// Kept for backward compatibility
export interface HistoryResponse {
  history: HistoryMessage[];
}

File: rag_ui/backend/models/history.py
from pydantic import BaseModel
from typing import List


class Message(BaseModel):
    id: str
    role: str
    content: str
    timestamp: str


class Session(BaseModel):
    session_id: str
    title: str
    created_at: str
    updated_at: str
    message_count: int


class SessionsResponse(BaseModel):
    sessions: List[Session]


class SessionDetail(BaseModel):
    session_id: str
    title: str
    messages: List[Message]


# Kept for backward compatibility
class HistoryResponse(BaseModel):
    history: List[Message]
